snapcraft/internal/lxd.py

```diff
@@ -204,18 +218,83 @@ def _ensure_container(self):
     def _setup_project(self):
         self._ensure_mount(self._project_folder, self._source)
 
+    def _get_container_address(self):
+        network = self._get_container_status()['state']['network']['eth0']
```

Comment:
For the record, you'll never run into the scenario where there's no external network access because we probe connectivity via a Python snippet after creating the container. Though the device may have a different name, so I added support for it as well as a test case which I just noticed I didn't have.